models/ed/R/model2netcdf.ED2.R

```diff
@@ -940,9 +971,33 @@ put_E_values <- function(yr, nc_var, out, lat, lon, begins, ends, pft_names, ...
   
   # ----- fill list
   
+  ##### setup output time and time bounds
+  ## Create a date vector that contains each month of the model run (e.g. "2001-07-01" "2001-08-01" "2001-09-01"....)
+  ## and which is the correct length for each full or partial year
+  output_date_vector <- seq(lubridate::floor_date(begins,"month"), by = "month", length.out = dim(out[[1]])[1] )
+  ## Create a vector of the number of days in each month by year
+  ## (e.g. 31 31 30 31 30 31)
+  num_days_per_month <- as.vector(lubridate::days_in_month(output_date_vector))
```

Comment:
Do you need `as.vector` here? `seq` as written on the previous line should already give you a vector.

Reply:
Probably not, just put that there to be sure ;)

Reply:
Not my hill to die on, but FWIW, I'm not a huge fan of this kind of "just to be sure" coercion. If there is some ambiguity about the type, I prefer making the expected type(s) crystal clear, with a loud error thrown if they are not. Unexpected coercion can introduce all kinds of downstream bugs that could be hard to debug. 

```
output_date_vector <- seq(...)
if (!is.numeric(output_date_vector) | <other checks> ) {
  print(output_date_vector)
  PEcAn.logger::logger.severe("Not a numeric vector. Something went wrong...")
}
num_days_per_month <- days_in_month(...) ...
```

Again, not going to fight about this -- I just like talking about my feelings 😉